File: Modulo 2/Clase03/app/src/main/java/com/area51/clase03/MainActivity.java
package com.area51.clase03;

import android.content.Intent;
import android.support.design.widget.FloatingActionButton;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    private RecyclerView rvDatos;
    private FloatingActionButton fabAgregar;
    private PersonaAdapter adapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        rvDatos = findViewById(R.id.rvDatos);
        fabAgregar = findViewById(R.id.fabAgregar);
    }

    @Override
    protected void onStart() {
        super.onStart();

        MetodoRealm metodoRealm = new MetodoRealm();
        ArrayList<Persona> lista = new ArrayList<>(metodoRealm.obtenerTodos());
        adapter = new PersonaAdapter(lista, this);
        rvDatos.setLayoutManager(new LinearLayoutManager(this));
        rvDatos.setAdapter(adapter);
    }

    @Override
    protected void onResume() {
        super.onResume();

        fabAgregar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent =
                        new Intent(
                                MainActivity.this, RegistroActivity.class);
                startActivity(intent);
            }
        });
    }
}

File: Modulo 3/ClaseRecycler/app/src/main/java/com/area51/claserecycler/MainActivity.kt
package com.area51.claserecycler

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import com.area51.claserecycler.database.MetodoDBImpl
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    var lista: ArrayList<Persona>? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        lista = arrayListOf()
        //lista!!.add(PersonaEntidad("Johann", "Jara", "Masculino"))
    }

    override fun onStart() {
        super.onStart()

        if (lista != null)
            lista!!.clear()
        val database = MetodoDBImpl()
        val datos = database.listar()
        if (datos != null) {
            for (item: com.area51.claserecycler.entidades.PersonaEntidad in datos) {
                val persona = Persona(item.id, item.nombre!!, item.apellido!!, item.genero!!)
                lista!!.add(persona)
            }
        }

        val adapter = PersonaAdapter(this, lista!!)
        rvDatos.layoutManager = LinearLayoutManager(this)
        rvDatos.adapter = adapter
    }

    override fun onResume() {
        super.onResume()

        fabAgregar.setOnClickListener {
            val intent = Intent(this, RegistroActivity::class.java)
            startActivity(intent)
        }
    }
}
